CoWorkMobileApp/Controllers/SearchViewController.swift

```diff
@@ -6,26 +6,192 @@
 //
 
 import UIKit
+import Cartography
+import RealmSwift
 
-class SearchViewController: UIViewController {
-    
-    lazy var gradient: CAGradientLayer = {
-        let gradient = CAGradientLayer()
-        gradient.type = .axial
-        gradient.colors = [
-            ThemeColors.grayColor?.cgColor ?? UIColor.darkGray.cgColor,
-            UIColor.lightGray.cgColor,
-            UIColor.white.cgColor
-        ]
-        gradient.locations = [0, 0.25, 1]
-        return gradient
+class SearchViewController: UIViewController, UICollectionViewDelegate {
+    
+    private enum Section: Int, CaseIterable, CustomStringConvertible {
+        case allResults
+        
+        var description: String {
+            switch self {
+            case .allResults: return "All Results"
+            }
+        }
+    }
+    
+    struct WorkspaceItem: Hashable {
+        var workspaceId: String?
+    }
+    
+    private var collectionView: UICollectionView!
+    private var diffableDataSource: UICollectionViewDiffableDataSource<Section, WorkspaceItem>!
+    
+    private var workspaceListItems: [WorkspaceItem] = []
+    private var workspaceNotificationToken: NotificationToken?
+    private var workspaceResults: Results<Workspace>?
+    
+    let viewModel = SearchViewModel()
+    
+    let searchController: UISearchController = {
```

Comment:
I would make this a lazy var as well.   Otherwise, all this initialization code will run init, and it doesn't need to.